ml-dhs-aws/src/main/java/com/marklogic/client/dhs/JobRunner.java

```diff
@@ -64,22 +87,64 @@ public void run(DatabaseClient client, InputStream csvRecords, InputStream jobCo
                         throwable.printStackTrace(System.out);
                      });
 
-         ObjectMapper objectMapper = new ObjectMapper();
-         ObjectNode jobControlObj = (ObjectNode) objectMapper.readTree(jobControl);
-         ObjectNode jobDef        = (ObjectNode) jobControlObj.get("job");
-         ObjectNode recordDef     = (ObjectNode) jobControlObj.get("record");
 
          ObjectLoader loader = new ObjectLoader(batcher, recordDef, getJobDirectory(), documentMetadata);
-         CSVConverter converter = new CSVConverter();
-
-// TODO: write before job document in /jobStart and /jobs/ID collections or send before job payload to DHF endpoint
+         CSVConverter converter = new CSVConverter();  
 
+         Iterator<ObjectNode> itr = converter.convertObject(csvRecords);
+         if(!itr.hasNext()) {
+        	 throw new MarkLogicIOException("No header found.");
+         }
+         ObjectNode csvNode = itr.next();
+         
+         String headerValue = csvNode.fieldNames().toString();
```

Comment:
Instead of serializing the list of field names to a string, does it make sense to construct an ArrayNode similar to the code in the com.marklogic.client.test.JacksonHandleTest unit test for the base Java API?